docs/iot/tutorials/gpio-input.md

```diff
@@ -0,0 +1,138 @@
+
+    ```dotnetcli
+    dotnet new console -o InputTutorial
+    ```
+
+1. [!INCLUDE [tutorial-add-packages](../includes/tutorial-add-packages.md)]
+1. Replace the contents of *Program.cs* with the following code:
+
+    :::code language="csharp" source="~/iot-samples/tutorials/InputTutorial/Program.cs" :::
+
+    In the preceding code:
+
+    - A [using declaration](../../csharp/whats-new/csharp-8.md#using-declarations) creates an instance of `GpioController`. The `using` declaration ensures the object is disposed and hardware resources are released properly.
+    - GPIO pin 21 is opened with  `PinMode.InputPullUp`.
+        - This opens the pin with a *PullUp* resistor engaged. In this mode, when the pin is connected to ground, it will return `PinValue.Low`. When the pin is disconnected from ground and the circuit is open, the pin returns `PinValue.High`.
+    - The initial status is written to a console using a ternary expression. The pin's current state is read with `Read()`. If it's `PinValue.High`, it writes the `alert` string to the console. Otherwise, it writes the `ready` string.
+    - `RegisterCallbackForPinValueChangedEvent()` registers an anonymous callback function for both the `PinEventTypes.Rising` and `PinEventTypes.Falling` events on the pin. These events correspond to pin states of `PinValue.High` and `PinValue.Low`, respectively.
+    - The callback function delegates to a method called `onPinEvent`. `onPinEvent` uses another ternary expression that also writes the corresponding `alert` or `ready` strings.
+    - The main thread sleeps indefinitely while waiting for pin events.
+
+1. [!INCLUDE [tutorial-build](../includes/tutorial-build.md)]
+1. [!INCLUDE [tutorial-deploy](../includes/tutorial-deploy.md)]
+1. Run the app on the Raspberry Pi by switching to the deployment directory and running the executable.
+
+    ```bash
+    ./InputTutorial
+    ```
+
+    The app displays text similar to the following:
+
+    ```console
+    Initial status (05/10/2022 15:59:25): Ready ✅
+    ```
+
+1. Disconnect pin 21 from ground. The console displays text similar to the following:
+
+    ```console
+    (05/10/2022 15:59:59) ALERT! 🚨
+    ```
+
+1. Reconnect pin 21 and ground. The console displays text similar to the following:
+
+    ```console
+    (05/10/2022 16:00:25) Ready ✅
+    ```
+
+1. Terminate the program by pressing <kbd>Ctrl</kbd>+<kbd>C</kbd>.
+
+Congratulations! You've used GPIO to detect input! There are many uses for this type of input. This example can be used with any scenario where a switch connects or breaks a circuit. Here's an example using it with a magnetic reed switch, which is often used to detect open doors or windows.
+
+:::image type="content" source="https://via.placeholder.com/600x400.png" alt-text="Animated GIF showing magnetic reed switch":::
+
+## Laser tripwire
+
+A similar application involves laser tripwires. Building a laser tripwire requires the following additional components:
```

Comment:
Let's transition a bit from the previous example.

```suggestion
Extending the previous example concept a bit further, let's take a look at how this could be applied to creating a laser tripwire. Building a laser tripwire requires the following additional components:
```